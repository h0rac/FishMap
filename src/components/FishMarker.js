/* global require */

import React, { Component } from 'react';
import { Image } from 'react-native';
import MapView from 'react-native-maps';
import PropTypes from 'prop-types'

class FishMarker extends Component {
  render() {
    const region = {
      latitude: this.props.marker.latitude,
      longitude: this.props.marker.longitude,
    };
    return (
      <MapView.Marker 
      key={this.props.index}
      coordinate={region}
      title={this.props.marker.title}
      onPress={this.props.callbackPress}
      identifier={this.props.title}>
      <Image source={require('../assets/perch.png')}  style={{width: 20, height: 20}}/>
      </MapView.Marker>
    );
  }
}

FishMarker.propTypes = {
  marker: PropTypes.object,
  callbackPress: PropTypes.func,
  title: PropTypes.string,
  index: PropTypes.number
}

export default FishMarker;
