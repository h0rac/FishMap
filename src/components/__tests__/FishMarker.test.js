import 'react-native';
import React from 'react';
import FishMarker from '../FishMarker';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
describe('Testing FishMarker component', () => {

	let tree;
	beforeEach(() => {
        const marker =  {
            latitude: 54.475408,
            longitude: 18.263086
        }
        tree = shallow(
		<FishMarker marker={marker}/>);
	});
	it('renders correctly', () => {
		expect(tree).toMatchSnapshot();
	});
})