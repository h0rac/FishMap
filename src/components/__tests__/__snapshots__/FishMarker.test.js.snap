// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FishMarker component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FishMarker
    marker={
      Object {
        "latitude": 54.475408,
        "longitude": 18.263086,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Image
        source={
          Object {
            "testUri": "../../../src/assets/perch.png",
          }
        }
        style={
          Object {
            "height": 20,
            "width": 20,
          }
        }
      />,
      "coordinate": Object {
        "latitude": 54.475408,
        "longitude": 18.263086,
      },
      "identifier": undefined,
      "onPress": [Function],
      "title": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "source": Object {
          "testUri": "../../../src/assets/perch.png",
        },
        "style": Object {
          "height": 20,
          "width": 20,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Image
          source={
            Object {
              "testUri": "../../../src/assets/perch.png",
            }
          }
          style={
            Object {
              "height": 20,
              "width": 20,
            }
          }
        />,
        "coordinate": Object {
          "latitude": 54.475408,
          "longitude": 18.263086,
        },
        "identifier": undefined,
        "onPress": [Function],
        "title": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "source": Object {
            "testUri": "../../../src/assets/perch.png",
          },
          "style": Object {
            "height": 20,
            "width": 20,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
